test-requirements.txt

```diff
@@ -1,6 +1,21 @@
-pytest
-coverage
-pytest-cov
-requests_mock
-codecov
+coverage~=4.3.4
+coverage-env-plugin~=0.1
+coverage-config-reload-plugin~=0.2
+codecov~=2.0.5
+moban~=0.0.9
+packaging~=16.8
+pytest~=3.6.1
+pytest-cov~=2.4
+pytest-env~=0.6.0
+pytest-error-for-skips~=1.0
+pytest-instafail~=0.3.0
+pytest-mock~=1.1
+pytest-profiling~=1.3.0
+pytest-reorder~=0.1.0
+pytest-reqs~=0.0.6
+pytest-timeout~=1.3.0
+pytest-xdist~=1.15
+requests-mock~=1.2
+pip!=9.0.2, !=10.0.*
+wheel~=0.29
```

Comment:
What is the need of all these extra `pytest-*` deps, isn't `pytest~=3.6.1` supposed to be enough?

Reply:
These are all the lovely stuff which work on coala/coala repo, and the template used is being deployed to every other coala repo as well so we can use them there.
pip caching means the extra setup time is negligible.  And the plugins all should do nothing unless enabled in `setup.cfg`.

After this, you can enable profiling -- see https://github.com/coala/coala/pull/5548 , which also shows time spent in pytest plugins so enabling other features can be 'costed' and rejected if the plugin is too expensive.

Reply:
Woah :o Amazing !!